spec.html

```diff
@@ -9401,18 +9398,14 @@ <h1>[[DefineOwnProperty]] ( _P_, _Desc_ )</h1>
           1. If Type(_P_) is String, then
             1. Let _numericIndex_ be ! CanonicalNumericIndexString(_P_).
             1. If _numericIndex_ is not *undefined*, then
-              1. If IsInteger(_numericIndex_) is *false*, return *false*.
-              1. If _numericIndex_ = *-0*, return *false*.
-              1. If _numericIndex_ &lt; 0, return *false*.
-              1. Let _length_ be _O_.[[ArrayLength]].
-              1. If _numericIndex_ &ge; _length_, return *false*.
               1. If IsAccessorDescriptor(_Desc_) is *true*, return *false*.
               1. If _Desc_ has a [[Configurable]] field and if _Desc_.[[Configurable]] is *true*, return *false*.
               1. If _Desc_ has an [[Enumerable]] field and if _Desc_.[[Enumerable]] is *false*, return *false*.
               1. If _Desc_ has a [[Writable]] field and if _Desc_.[[Writable]] is *false*, return *false*.
               1. If _Desc_ has a [[Value]] field, then
                 1. Let _value_ be _Desc_.[[Value]].
                 1. Return ? IntegerIndexedElementSet(_O_, _numericIndex_, _value_).
+              1. If IsIntegerIndexValid(_O_, _numericIndex_) is *false*, return *false*.
```

Comment:
per @anba's comment, i think this needs to move up to line 9401?

Reply:
Yes, will move this back up.